mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
```

Comment:
I saw comments about git archive working from anywhere. That is how bzr and svn work. They turn out to be wrong for git. git dies with "not a repository" errors for any out-of-tree attempts to archive/export, such as our CI was doing with bzr.
Where bzr was happy with just a branch path and took care of checkout etc. git requires archive to be run from inside a clone repository, with a specific branch checkout.

So to retain the same snapshot result git requires two parameters to three commands where bzr took only one parameter to one command.

Your statement about tags being global is incorrect for the magical HEAD tags. A "git archive HEAD" on checkout master produces a bundle of revision=56174e91c whereas in the same repository clone after "checkout v4.0"  that git command produces a bundle of revision=ce53d9579.

That is why the branch is required separate from tag now on the script command line. To do anything beyond the default HEAD tag it cannot be assumed from the branch path like bzr was doing.